flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/RecoveredInputChannelTest.java

```diff
@@ -252,10 +255,8 @@ private void testConcurrentReadStateAndProcessAndRelease(boolean isRemote) throw
 						buffer.recycleBuffer();
 						numProcessedStates++;
 					}
-				} catch (IllegalStateException e) {
-					if (!e.getMessage().contains("Queried for a buffer after channel has been closed")) {
-						throw e;
-					}
+				} catch (Throwable t) {
+					assertTrue("The released channel is expected, but the actual exception is " + t.getMessage(), verifyRelease && inputChannel.isReleased());
```

Comment:
nit: you could keep the cause by throwing `new AssertionError(msg, t)`, which would preserve the stack trace as well.